File: java/example/src/global/hiber/example/ModemServiceClient.java
package global.hiber.example;

import global.hiber.api.grpc.modem.ModemApi.ListModemsRequest;
import global.hiber.api.grpc.modem.ModemApi.Modem;
import global.hiber.api.grpc.modem.ModemServiceGrpc;
import io.grpc.ManagedChannel;
import io.grpc.StatusRuntimeException;

import java.util.List;

class ModemServiceClient {
  private final ModemServiceGrpc.ModemServiceBlockingStub stub;

  ModemServiceClient(ManagedChannel channel, String token) {
    this.stub = ModemServiceGrpc
      .newBlockingStub(channel)
      .withCallCredentials(new Authenticated(token));
  }

  List<Modem> modems() {
    try {
      return stub.list(ListModemsRequest.getDefaultInstance()).getModemsList();
    } catch (StatusRuntimeException e) {
      System.out.println("GRPC call failed:");
      throw e;
    }
  }
}

File: kotlin/example/src/global/hiber/example/ModemServiceClient.kt
package global.hiber.example

import global.hiber.api.grpc.modem.ModemApi.ListModemsRequest
import global.hiber.api.grpc.modem.ModemApi.Modem
import global.hiber.api.grpc.modem.ModemServiceGrpc
import io.grpc.ManagedChannel
import io.grpc.StatusRuntimeException

class ModemServiceClient(val channel: ManagedChannel, val token: String) {
  private val stub: ModemServiceGrpc.ModemServiceBlockingStub =
    ModemServiceGrpc
      .newBlockingStub(channel)
      .withCallCredentials(Authenticated(token))

  fun modems(): List<Modem> =
    try {
      stub.list(ListModemsRequest.getDefaultInstance()).getModemsList()
    } catch (e: StatusRuntimeException) {
      System.out.println("GRPC call failed:")
      throw e
    }
}
